fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java

```diff
@@ -285,34 +282,54 @@ private static void transferLeader() throws InterruptedException {
                     while (true) {
                         try {
                             InetSocketAddress address = GlobalStateMgr.getCurrentState().getHaProtocol().getLeader();
-                            LOG.info("leader is transferred to {}", address);
-                            break;
+                            // wait for new leader to be ready
+                            if (isNewLeaderReady(address.getHostString())) {
+                                LOG.info("leader is transferred to {}", address);
+                                break;
+                            }
                         } catch (Exception e) {
-                            Thread.sleep(100L);
+                            Thread.sleep(300L);
                         }
                     }
                 }
 
                 GlobalStateMgr.getCurrentState().markLeaderTransferred();
             }
+        }
+    }
 
-            AgentTaskQueue.failForLeaderTransfer();
+    private static boolean isNewLeaderReady(String leaderHost) {
+        String accessibleHostPort = NetUtils.getHostPortInAccessibleFormat(leaderHost, Config.http_port);
+        String url = "http://" + accessibleHostPort
```

Comment:
// TODO, this is a hard of http protocol, in case https is required later on which is on the way.